website/pages/blog/graphql-yoga-worker.mdx

```diff
@@ -0,0 +1,408 @@
+  require Node.js
+
+## Motivation
+
+Cloudflare Workers is often used as a proxy on a CDN or as a backend for rendering Remix or Next.js.
+When it comes to APIs for retrieving and updating data from the front end, most of the time I think
+it is necessary to set up and build a separate backend server. I myself often set up a backend
+server in Node.js in this pattern, and when I do so, I choose Cloud Run as my first choice because
+it is easy to set up. Cloud Run itself is a great service, but it is human beings who become even
+more greedy.
+
+- Deployment speed needs to be faster.
+- I want to keep the price a little lower.
+
+The former is a container service, so it requires "container build" and then "container
+replacement", but "container replacement" as well as "container build" takes a lot of time there. No
+matter how fast you do these two, it will take 5 to 10 minutes. The latter means minimizing cold
+standby (startup speed) and asynchronous means using the minimum number of containers, Always CPU,
+etc. That will cost a certain amount of money. I still think Cloud Run is cheaper, though.
+
+I believe that Cloudflare Workers can fulfill these requests, but if it is actually necessary to set
+up an API server, a database connection is required. if it is something that can communicate via
+PostgREST API such as [Supabase](https://supabase.com/), Cloudflare Workers can connect to the
+database via the PostgREST API such as Supabase, but making a direct connection to the database has
+not been possible until now. However, this situation has changed with the recent announcement of
+support for connections over TCP.
+
+https://blog.cloudflare.com/workers-tcp-socket-api-connect-databases/
+
+I think that just because it is possible to connect via TCP, the advantage of the edge location is
+lost a bit, because if an application running on the edge connects to a database in another location
+to retrieve data, there is an overhead for that connection. So I'm waiting with anticipation for the
+[Cloudflare D1](https://developers.cloudflare.com/d1/) to come out in the future, as it will
+maximize the advantages of that location. Cloudflare Workers is as great a service as Cloud Run, so
+the purpose of this article is to actually run it as a backend API server on top of Cloudflare
+Workers.
+
+## Architectures
+
+Let's proceed under the assumption that the backend API will be prepared as a GraphQL server, and I
+will write down what to use to build it.
+
+### Cloudfalre Workers
+
+https://workers.cloudflare.com/
+
+The first assumption is this. The choice of Cloudflare Workers as a service or runtime narrows the
+technologies that can be used in an application. Therefore, it is written as an assumption.
+
+### Web Server
+
+https://the-guild.dev/graphql/yoga-server
+
+Web Servers that work with Cloudflare Workers are pretty limited. This is because Node.js does not
```

Comment:
```suggestion
Many existing/older Web Servers have limited support for Cloudflare Workers. That's because they were built at a time to only support Node.js. 

Newer Web Server frameworks like [feTS](https://the-guild.dev/openapi/fets/server/quick-start) or [Hono](https://hono.dev/) are modern and have better support for Cloudflare Workers.

In our case, we will use [GraphQL Yoga](https://the-guild.dev/graphql/yoga-server) because it support GraphQL and also uses the same powerful [@whatwg-node library](https://github.com/ardatan/whatwg-node) as [feTS](https://the-guild.dev/openapi/fets/server/quick-start), to support [any Javascript runtime](https://the-guild.dev/blog/fetch-for-servers), including Cloudflare Workers.
```